src/model.h

```diff
@@ -146,6 +146,48 @@ class Model {
     return stats_aggregator_;
   }
 
+  // Directly call from C API, so arguments are in the same style
+  void SetMemoryUsage(BufferAttributes** memory_usage, uint32_t usage_size)
```

Comment:
This could probably be implicitly converted to a `boost::span<BufferAttributes*>` if you wanted to not use the C style in implementation.